js/canvas.js

```diff
@@ -0,0 +1,367 @@
+import { arrayOfRectangles } from './rectangles.js';
+import { canvasWindowIndent, startedRectangleFillColor, redColorInHex, stickyDistance } from './shared/constants.js';
+
+export class Canvas {
+  constructor(selectedCanvas) {
+    this.canvas = selectedCanvas;
+    this.ctx = this.canvas.getContext('2d');
+    this.draggingRect;
+    this.isDragOk = false;
+    this.offsetX;
+    this.offsetY;
+  }
+
+  start() {
+    this.setSize();
+    this.setOffsets_X_and_Y();
+    this.drawRectangles();
+    this.canvas.addEventListener('mousedown', this.mouseDownHandler.bind(this));
+    this.canvas.addEventListener('mousemove', this.mouseMoveHandler.bind(this));
+    this.canvas.addEventListener('mouseup', this.mouseUpHandler.bind(this));
+  }
+
+  setSize() {
+    const { innerWidth: width, innerHeight: height } = window;
+
+    this.canvas.width = width - canvasWindowIndent;
+    this.canvas.height = height - canvasWindowIndent;
+  }
+
+  setOffsets_X_and_Y() {
+    const domRect = canvas.getBoundingClientRect();
+    const { left, top } = domRect;
+
+    this.offsetX = left;
+    this.offsetY = top;
+  }
+
+  drawRectangles() {
+    this.clear();
+
+    arrayOfRectangles.forEach(rect => {
+      if (rect.isSticked && !rect.isCrossed) {
+        const { stickedX, stickedY, width, height } = rect;
+
+        this.ctx.fillStyle = rect.fillColor;
+        this.ctx.fillRect(stickedX, stickedY, width, height);
+      } else {
+        const { x, y, width, height } = rect;
+
+        this.ctx.fillStyle = rect.fillColor;
+        this.ctx.fillRect(x, y, width, height);
+      }
+    });
+  }
+
+  clear() {
+    const { width, height } = this.canvas;
+
+    this.ctx.clearRect(0, 0, width, height);
+  }
+
+  isRectangle_a_and_b_crossed(rectA, rectB) {
```

Comment:
Don't need this methid inside class because you don't call any class's method (don't use `this` inside this method). Suggest to move outside of class... maybe in some helper function